modelling/src/test/java/org/axonframework/modelling/SimpleStateManagerTest.java

```diff
@@ -0,0 +1,431 @@
+                                                            .register(repository)
+                                                            .register(repository2)
+                                                            .build();
+
+        // when
+        var result = stateManager.registeredIdsFor(MyFirstImplementingClass.class);
+
+        // then
+        assertEquals(2, result.size());
+        assertTrue(result.contains(String.class));
+        assertTrue(result.contains(Integer.class));
+    }
+
+    @Nested
+    class PolymorphicRepositories {
+
+        @Test
+        void canRequestMoreSpecificEntityThanDefinedInRepositoryIfTypeMatches() {
+            // given
+            SimpleRepository<String, MySuperClass> repository = new SimpleRepository<>(
+                    String.class,
+                    MySuperClass.class,
+                    (id, context) -> CompletableFuture.completedFuture(new MyFirstImplementingClass()),
+                    (id, entity, context) -> CompletableFuture.completedFuture(null)
+            );
+            SimpleStateManager stateManager = SimpleStateManager.builder("test")
+                                                                .register(repository)
+                                                                .build();
+
+            // when
+            var result = stateManager.loadManagedEntity(MyFirstImplementingClass.class,
+                                                        "42",
+                                                        new StubProcessingContext()).join();
+
+            // then
+            assertNotNull(result.entity());
+        }
+
+        @Test
+        void canNotRequestLessSpecificEntityThanDefinedInRepository() {
+            // given
+            SimpleRepository<String, MyFirstImplementingClass> repository = new SimpleRepository<>(
+                    String.class,
+                    MyFirstImplementingClass.class,
+                    (id, context) -> CompletableFuture.completedFuture(new MyFirstImplementingClass()),
+                    (id, entity, context) -> CompletableFuture.completedFuture(null)
+            );
+            SimpleStateManager stateManager = SimpleStateManager.builder("test")
+                                                                .register(repository)
+                                                                .build();
+
+            // when & then
+            assertThrows(MissingRepositoryException.class,
+                         () -> stateManager.loadManagedEntity(MySuperClass.class, "42", new StubProcessingContext())
+                                           .join());
+        }
+
+        @Test
+        void canRequestAnEntityWithAMoreSpecificIdentifierThanDefinedByRepository() {
+            // given
+            SimpleRepository<MySuperClass, String> repository = new SimpleRepository<>(
+                    MySuperClass.class,
```

Comment:
I had to look at this test for a bit to make sure it does what it should. Although I understand you can use the same classes as with the entity-type validation, it makes it (at least for me) hard to reason whether the test is doing the right thing :see_no_evil:

Reply:
Yes, you are right. Adjusting it to read:

```java
SimpleRepository<MySuperId, MySuperEntity> repository = new SimpleRepository<>(
                    MySuperId.class,
                    MySuperEntity.class,
                    (id, context) -> CompletableFuture.completedFuture(new MySuperEntity()),
                    (id, entity, context) -> CompletableFuture.completedFuture(null)
            );
```

Reply:
Thanks for that man!